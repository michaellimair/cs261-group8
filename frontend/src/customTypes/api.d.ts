type ApiBadRequestErrorKeys<T> = keyof T | 'non_field_errors';

export type IApiBadRequestErrorData<U> = {
  [key in ApiBadRequestErrorKeys<U>]: string[];
};
