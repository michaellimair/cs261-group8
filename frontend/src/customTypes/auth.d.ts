import { IApiBadRequestErrorData } from './api';

export interface IUser {
  username: string;
  email: string;
  first_name: string;
  last_name: string;
}

export interface IRegistration {
  first_name: string;
  last_name: string;
  username: string;
  email: string;
  password: string;
  verify_password: string;
}

export interface ILogin {
  username: string;
  password: string;
}

export interface ILoginResult {
  user: Pick<IUser, 'username'>;
  token: string;
  expiry: Date;
}

/**
 * Login error type to use if the error is a Bad Request error,
 * possibly due to the wrong username or password.
 */
export type ILoginError = IApiBadRequestErrorData<ILogin>;
