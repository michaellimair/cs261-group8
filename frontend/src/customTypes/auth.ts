import { IApiBadRequestErrorData } from './api';

export enum JobTitle {
  ANALYST = 'anlst',
  ASSOCIATE = 'assoc',
  ASSISTANT_VICE_PRESIDENT = 'avp',
  VICE_PRESIDENT = 'vp',
  DIRECTOR = 'dir',
  MANAGING_DIRECTOR = 'md',
}

export interface IBusinessArea {
  name: string;
  label: string;
}

export interface IUserProfile {
  completed: boolean;
  pronoun: string | null;
  years_experience: number | null;
  title: JobTitle | null;
  business_area: IBusinessArea | null;
}

export enum UserGroup {
  MENTOR = 'mentor',
  MENTEE = 'mentee',
}

export interface IUserGroup {
  id: number;
  name: UserGroup;
}

export interface IUser {
  id: number;
  username: string;
  email: string;
  first_name: string;
  last_name: string;
  profile: IUserProfile;
  groups: IUserGroup[];
}

export interface IRegistration {
  first_name: string;
  last_name: string;
  username: string;
  email: string;
  password: string;
  verify_password: string;
}

export interface ILogin {
  username: string;
  password: string;
}

export interface ILoginResult {
  user: Pick<IUser, 'username'>;
  token: string;
  expiry: Date;
}

/**
 * Login error type to use if the error is a Bad Request error,
 * possibly due to the wrong username or password.
 */
export type ILoginError = IApiBadRequestErrorData<Partial<ILogin>>;

export type IRegistrationError = IApiBadRequestErrorData<Partial<IRegistration>>;
