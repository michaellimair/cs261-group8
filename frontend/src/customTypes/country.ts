export interface ICountry {
  alpha_2: string;
  alpha_3: string;
  flag: string;
  name: string;
  numeric: string;
}
