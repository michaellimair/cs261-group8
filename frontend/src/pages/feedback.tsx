import { FC } from 'react';

const FeedbackPage: FC = () => (
  <div>
    hello
  </div>
);

export default FeedbackPage;
