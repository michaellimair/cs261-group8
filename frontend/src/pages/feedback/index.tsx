import { FC } from 'react';
import { useTranslation } from 'react-i18next';
import { Heading, HStack, Stack } from '@chakra-ui/react';
import FeedbackList from 'components/Feedback/FeedbackList';
import CreateFeedbackLink from 'components/Feedback/CreateFeedbackLink';
import { httpClient } from 'api';
import { useQuery } from 'react-query';

const FeedbackPage: FC = () => {
  const { t } = useTranslation();
  const { data, isLoading } = useQuery(
    ['feedback', 'list'],
    () => httpClient.feedback.listFeedback(),
  );

  return (
    <Stack>
      <HStack justifyContent="space-between" alignItems="center">
        <Heading as="h2" size="md">
          {t('dashboard.feedback.your_feedbacks')}
        </Heading>
        <CreateFeedbackLink />
      </HStack>
      <FeedbackList isLoading={isLoading} feedbacks={data} />
    </Stack>
  );
};

export default FeedbackPage;
