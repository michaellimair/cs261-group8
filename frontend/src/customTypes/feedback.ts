import { IUser } from './auth';

export enum FeedbackType {
  BUG = 'bug',
  IMPROVEMENT = 'improvement',
  QUESTION = 'question',
}

export interface IFeedbackReply {
  id: number;
  content: string;
  created: Date;
  updated: Date;
  admin: IUser;
}

export interface IFeedback {
  id: number;
  type: FeedbackType;
  content: string;
  created: Date;
  updated: Date;
  reply: null | IFeedbackReply;
}
