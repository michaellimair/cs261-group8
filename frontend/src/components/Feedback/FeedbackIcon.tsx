import { FeedbackType } from 'customTypes/feedback';
import { FC } from 'react';
import {
  Tag, TagLabel, TagLeftIcon, ThemingProps,
} from '@chakra-ui/react';
import { FcHighPriority, FcIdea } from 'react-icons/fc';
import { GrCircleQuestion } from 'react-icons/gr';
import { useTranslation } from 'react-i18next';

export interface IFeedbackIconProps {
  type: FeedbackType;
}

const useFeedbackIcon = (type: FeedbackType) => {
  if (type === FeedbackType.IMPROVEMENT) return FcIdea;
  if (type === FeedbackType.BUG) return FcHighPriority;
  return GrCircleQuestion;
};

const useFeedbackColor = (type: FeedbackType): ThemingProps['colorScheme'] => {
  if (type === FeedbackType.IMPROVEMENT) return 'blue';
  if (type === FeedbackType.BUG) return 'red';
  return 'cyan';
};

const FeedbackIcon: FC<IFeedbackIconProps> = ({ type }) => {
  const { t } = useTranslation();

  return (
    <Tag size="lg" variant="outline" colorScheme={useFeedbackColor(type)}>
      <TagLeftIcon boxSize="12px" as={useFeedbackIcon(type)} />
      <TagLabel>{t(`dashboard.feedback.${type}`)}</TagLabel>
    </Tag>
  );
};

export default FeedbackIcon;
