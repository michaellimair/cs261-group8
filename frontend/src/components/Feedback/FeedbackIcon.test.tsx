import { render } from '@testing-library/react';
import { FeedbackType } from 'customTypes/feedback';
import FeedbackIcon from './FeedbackIcon';

describe('components/Feedback/FeedbackIcon', () => {
  it('renders bug icon when feedback type is a bug', () => {
    const result = render(<FeedbackIcon type={FeedbackType.BUG} />);

    expect(result).toMatchSnapshot();
  });

  it('renders idea icon when feedback type is improvement', () => {
    const result = render(<FeedbackIcon type={FeedbackType.IMPROVEMENT} />);

    expect(result).toMatchSnapshot();
  });

  it('falls back to rendering question icon for other question types', () => {
    const result = render(<FeedbackIcon type={FeedbackType.QUESTION} />);

    expect(result).toMatchSnapshot();
  });
});
