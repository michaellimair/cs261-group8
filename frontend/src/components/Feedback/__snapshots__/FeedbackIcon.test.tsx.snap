// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components/Feedback/FeedbackIcon falls back to rendering question icon for other question types 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div
      id="chakra-toast-portal"
    >
      <ul
        id="chakra-toast-manager-top"
        style="position: fixed; z-index: 5500; pointer-events: none; display: flex; flex-direction: column; margin: 0px auto;"
      />
      <ul
        id="chakra-toast-manager-top-left"
        style="position: fixed; z-index: 5500; pointer-events: none; display: flex; flex-direction: column;"
      />
      <ul
        id="chakra-toast-manager-top-right"
        style="position: fixed; z-index: 5500; pointer-events: none; display: flex; flex-direction: column;"
      />
      <ul
        id="chakra-toast-manager-bottom-left"
        style="position: fixed; z-index: 5500; pointer-events: none; display: flex; flex-direction: column;"
      />
      <ul
        id="chakra-toast-manager-bottom"
        style="position: fixed; z-index: 5500; pointer-events: none; display: flex; flex-direction: column; margin: 0px auto;"
      />
      <ul
        id="chakra-toast-manager-bottom-right"
        style="position: fixed; z-index: 5500; pointer-events: none; display: flex; flex-direction: column;"
      />
    </div>
    <div>
      <span
        class="css-1qm7lvz"
      >
        <svg
          class="chakra-icon css-1azbes1"
          fill="currentColor"
          focusable="false"
          height="1em"
          stroke="currentColor"
          stroke-width="0"
          viewBox="0 0 24 24"
          width="1em"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M12,22 C17.5228475,22 22,17.5228475 22,12 C22,6.4771525 17.5228475,2 12,2 C6.4771525,2 2,6.4771525 2,12 C2,17.5228475 6.4771525,22 12,22 Z M12,15 L12,14 C12,13 12,12.5 13,12 C14,11.5 15,11 15,9.5 C15,8.5 14,7 12,7 C10,7 9,8.26413718 9,10 M12,16 L12,18"
            fill="none"
            stroke="#000"
            stroke-width="2"
          />
        </svg>
        <span
          class="css-8uhtka"
        >
          dashboard.feedback.question
        </span>
      </span>
    </div>
  </body>,
  "container": <div>
    <span
      class="css-1qm7lvz"
    >
      <svg
        class="chakra-icon css-1azbes1"
        fill="currentColor"
        focusable="false"
        height="1em"
        stroke="currentColor"
        stroke-width="0"
        viewBox="0 0 24 24"
        width="1em"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M12,22 C17.5228475,22 22,17.5228475 22,12 C22,6.4771525 17.5228475,2 12,2 C6.4771525,2 2,6.4771525 2,12 C2,17.5228475 6.4771525,22 12,22 Z M12,15 L12,14 C12,13 12,12.5 13,12 C14,11.5 15,11 15,9.5 C15,8.5 14,7 12,7 C10,7 9,8.26413718 9,10 M12,16 L12,18"
          fill="none"
          stroke="#000"
          stroke-width="2"
        />
      </svg>
      <span
        class="css-8uhtka"
      >
        dashboard.feedback.question
      </span>
    </span>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`components/Feedback/FeedbackIcon renders bug icon when feedback type is a bug 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div
      id="chakra-toast-portal"
    >
      <ul
        id="chakra-toast-manager-top"
        style="position: fixed; z-index: 5500; pointer-events: none; display: flex; flex-direction: column; margin: 0px auto;"
      />
      <ul
        id="chakra-toast-manager-top-left"
        style="position: fixed; z-index: 5500; pointer-events: none; display: flex; flex-direction: column;"
      />
      <ul
        id="chakra-toast-manager-top-right"
        style="position: fixed; z-index: 5500; pointer-events: none; display: flex; flex-direction: column;"
      />
      <ul
        id="chakra-toast-manager-bottom-left"
        style="position: fixed; z-index: 5500; pointer-events: none; display: flex; flex-direction: column;"
      />
      <ul
        id="chakra-toast-manager-bottom"
        style="position: fixed; z-index: 5500; pointer-events: none; display: flex; flex-direction: column; margin: 0px auto;"
      />
      <ul
        id="chakra-toast-manager-bottom-right"
        style="position: fixed; z-index: 5500; pointer-events: none; display: flex; flex-direction: column;"
      />
    </div>
    <div>
      <span
        class="css-1qm7lvz"
      >
        <svg
          class="chakra-icon css-1azbes1"
          enable-background="new 0 0 48 48"
          fill="currentColor"
          focusable="false"
          height="1em"
          stroke="currentColor"
          stroke-width="0"
          version="1"
          viewBox="0 0 48 48"
          width="1em"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M21.2,44.8l-18-18c-1.6-1.6-1.6-4.1,0-5.7l18-18c1.6-1.6,4.1-1.6,5.7,0l18,18c1.6,1.6,1.6,4.1,0,5.7l-18,18 C25.3,46.4,22.7,46.4,21.2,44.8z"
            fill="#F44336"
          />
          <path
            d="M21.6,32.7c0-0.3,0.1-0.6,0.2-0.9c0.1-0.3,0.3-0.5,0.5-0.7c0.2-0.2,0.5-0.4,0.8-0.5s0.6-0.2,1-0.2 s0.7,0.1,1,0.2c0.3,0.1,0.6,0.3,0.8,0.5c0.2,0.2,0.4,0.4,0.5,0.7c0.1,0.3,0.2,0.6,0.2,0.9s-0.1,0.6-0.2,0.9s-0.3,0.5-0.5,0.7 c-0.2,0.2-0.5,0.4-0.8,0.5c-0.3,0.1-0.6,0.2-1,0.2s-0.7-0.1-1-0.2s-0.5-0.3-0.8-0.5c-0.2-0.2-0.4-0.4-0.5-0.7S21.6,33.1,21.6,32.7z M25.8,28.1h-3.6L21.7,13h4.6L25.8,28.1z"
            fill="#fff"
          />
        </svg>
        <span
          class="css-8uhtka"
        >
          dashboard.feedback.bug
        </span>
      </span>
    </div>
  </body>,
  "container": <div>
    <span
      class="css-1qm7lvz"
    >
      <svg
        class="chakra-icon css-1azbes1"
        enable-background="new 0 0 48 48"
        fill="currentColor"
        focusable="false"
        height="1em"
        stroke="currentColor"
        stroke-width="0"
        version="1"
        viewBox="0 0 48 48"
        width="1em"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M21.2,44.8l-18-18c-1.6-1.6-1.6-4.1,0-5.7l18-18c1.6-1.6,4.1-1.6,5.7,0l18,18c1.6,1.6,1.6,4.1,0,5.7l-18,18 C25.3,46.4,22.7,46.4,21.2,44.8z"
          fill="#F44336"
        />
        <path
          d="M21.6,32.7c0-0.3,0.1-0.6,0.2-0.9c0.1-0.3,0.3-0.5,0.5-0.7c0.2-0.2,0.5-0.4,0.8-0.5s0.6-0.2,1-0.2 s0.7,0.1,1,0.2c0.3,0.1,0.6,0.3,0.8,0.5c0.2,0.2,0.4,0.4,0.5,0.7c0.1,0.3,0.2,0.6,0.2,0.9s-0.1,0.6-0.2,0.9s-0.3,0.5-0.5,0.7 c-0.2,0.2-0.5,0.4-0.8,0.5c-0.3,0.1-0.6,0.2-1,0.2s-0.7-0.1-1-0.2s-0.5-0.3-0.8-0.5c-0.2-0.2-0.4-0.4-0.5-0.7S21.6,33.1,21.6,32.7z M25.8,28.1h-3.6L21.7,13h4.6L25.8,28.1z"
          fill="#fff"
        />
      </svg>
      <span
        class="css-8uhtka"
      >
        dashboard.feedback.bug
      </span>
    </span>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`components/Feedback/FeedbackIcon renders idea icon when feedback type is improvement 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div
      id="chakra-toast-portal"
    >
      <ul
        id="chakra-toast-manager-top"
        style="position: fixed; z-index: 5500; pointer-events: none; display: flex; flex-direction: column; margin: 0px auto;"
      />
      <ul
        id="chakra-toast-manager-top-left"
        style="position: fixed; z-index: 5500; pointer-events: none; display: flex; flex-direction: column;"
      />
      <ul
        id="chakra-toast-manager-top-right"
        style="position: fixed; z-index: 5500; pointer-events: none; display: flex; flex-direction: column;"
      />
      <ul
        id="chakra-toast-manager-bottom-left"
        style="position: fixed; z-index: 5500; pointer-events: none; display: flex; flex-direction: column;"
      />
      <ul
        id="chakra-toast-manager-bottom"
        style="position: fixed; z-index: 5500; pointer-events: none; display: flex; flex-direction: column; margin: 0px auto;"
      />
      <ul
        id="chakra-toast-manager-bottom-right"
        style="position: fixed; z-index: 5500; pointer-events: none; display: flex; flex-direction: column;"
      />
    </div>
    <div>
      <span
        class="css-1qm7lvz"
      >
        <svg
          class="chakra-icon css-1azbes1"
          enable-background="new 0 0 48 48"
          fill="currentColor"
          focusable="false"
          height="1em"
          stroke="currentColor"
          stroke-width="0"
          version="1"
          viewBox="0 0 48 48"
          width="1em"
          xmlns="http://www.w3.org/2000/svg"
        >
          <circle
            cx="24"
            cy="22"
            fill="#FFF59D"
            r="20"
          />
          <path
            d="M37,22c0-7.7-6.6-13.8-14.5-12.9c-6,0.7-10.8,5.5-11.4,11.5c-0.5,4.6,1.4,8.7,4.6,11.3 c1.4,1.2,2.3,2.9,2.3,4.8V37h12v-0.1c0-1.8,0.8-3.6,2.2-4.8C35.1,29.7,37,26.1,37,22z"
            fill="#FBC02D"
          />
          <path
            d="M30.6,20.2l-3-2c-0.3-0.2-0.8-0.2-1.1,0L24,19.8l-2.4-1.6c-0.3-0.2-0.8-0.2-1.1,0l-3,2 c-0.2,0.2-0.4,0.4-0.4,0.7s0,0.6,0.2,0.8l3.8,4.7V37h2V26c0-0.2-0.1-0.4-0.2-0.6l-3.3-4.1l1.5-1l2.4,1.6c0.3,0.2,0.8,0.2,1.1,0 l2.4-1.6l1.5,1l-3.3,4.1C25.1,25.6,25,25.8,25,26v11h2V26.4l3.8-4.7c0.2-0.2,0.3-0.5,0.2-0.8S30.8,20.3,30.6,20.2z"
            fill="#FFF59D"
          />
          <circle
            cx="24"
            cy="44"
            fill="#5C6BC0"
            r="3"
          />
          <path
            d="M26,45h-4c-2.2,0-4-1.8-4-4v-5h12v5C30,43.2,28.2,45,26,45z"
            fill="#9FA8DA"
          />
          <g
            fill="#5C6BC0"
          >
            <path
              d="M30,41l-11.6,1.6c0.3,0.7,0.9,1.4,1.6,1.8l9.4-1.3C29.8,42.5,30,41.8,30,41z"
            />
            <polygon
              points="18,38.7 18,40.7 30,39 30,37"
            />
          </g>
        </svg>
        <span
          class="css-8uhtka"
        >
          dashboard.feedback.improvement
        </span>
      </span>
    </div>
  </body>,
  "container": <div>
    <span
      class="css-1qm7lvz"
    >
      <svg
        class="chakra-icon css-1azbes1"
        enable-background="new 0 0 48 48"
        fill="currentColor"
        focusable="false"
        height="1em"
        stroke="currentColor"
        stroke-width="0"
        version="1"
        viewBox="0 0 48 48"
        width="1em"
        xmlns="http://www.w3.org/2000/svg"
      >
        <circle
          cx="24"
          cy="22"
          fill="#FFF59D"
          r="20"
        />
        <path
          d="M37,22c0-7.7-6.6-13.8-14.5-12.9c-6,0.7-10.8,5.5-11.4,11.5c-0.5,4.6,1.4,8.7,4.6,11.3 c1.4,1.2,2.3,2.9,2.3,4.8V37h12v-0.1c0-1.8,0.8-3.6,2.2-4.8C35.1,29.7,37,26.1,37,22z"
          fill="#FBC02D"
        />
        <path
          d="M30.6,20.2l-3-2c-0.3-0.2-0.8-0.2-1.1,0L24,19.8l-2.4-1.6c-0.3-0.2-0.8-0.2-1.1,0l-3,2 c-0.2,0.2-0.4,0.4-0.4,0.7s0,0.6,0.2,0.8l3.8,4.7V37h2V26c0-0.2-0.1-0.4-0.2-0.6l-3.3-4.1l1.5-1l2.4,1.6c0.3,0.2,0.8,0.2,1.1,0 l2.4-1.6l1.5,1l-3.3,4.1C25.1,25.6,25,25.8,25,26v11h2V26.4l3.8-4.7c0.2-0.2,0.3-0.5,0.2-0.8S30.8,20.3,30.6,20.2z"
          fill="#FFF59D"
        />
        <circle
          cx="24"
          cy="44"
          fill="#5C6BC0"
          r="3"
        />
        <path
          d="M26,45h-4c-2.2,0-4-1.8-4-4v-5h12v5C30,43.2,28.2,45,26,45z"
          fill="#9FA8DA"
        />
        <g
          fill="#5C6BC0"
        >
          <path
            d="M30,41l-11.6,1.6c0.3,0.7,0.9,1.4,1.6,1.8l9.4-1.3C29.8,42.5,30,41.8,30,41z"
          />
          <polygon
            points="18,38.7 18,40.7 30,39 30,37"
          />
        </g>
      </svg>
      <span
        class="css-8uhtka"
      >
        dashboard.feedback.improvement
      </span>
    </span>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
